pkg/codeconfig/codeconfig.go

```diff
@@ -83,18 +83,38 @@ func imageNameFromExt(ext string) string {
 
 // Collect - Collects information about all functions for a nitric stack
 func (c *codeConfig) Collect() error {
+	wg := sync.WaitGroup{}
+	errs := sync.Map{}
+
 	for _, f := range c.files {
-		rel, err := filepath.Rel(c.stackPath, f)
-		if err != nil {
-			return err
-		}
+		wg.Add(1)
 
-		err = c.collectOne(rel)
-		if err != nil {
-			return err
-		}
+		// run files in parallel
+		go func(file string) {
+			defer wg.Done()
+			rel, err := filepath.Rel(c.stackPath, file)
+			if err != nil {
+				errs.Store(rel, err)
+				return
+			}
+
+			err = c.collectOne(rel)
+			if err != nil {
+				errs.Store(rel, err)
+				return
+			}
+		}(f)
 	}
-	return nil
+
+	wg.Wait()
+
+	errList := utils.NewErrorList()
```

Comment:
couldn't you just add a lock in the errorList.Add()?

Reply:
@asalkeld sure can, wasn't sure if we'd want to make the list itself threadsafe. Will update this now.